pkg/executor/adapter_slow_log.go

```diff
@@ -0,0 +1,452 @@
+// Copyright 2025 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package executor
+
+import (
+	"bytes"
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/pingcap/tidb/pkg/keyspace"
+	"github.com/pingcap/tidb/pkg/parser/ast"
+	"github.com/pingcap/tidb/pkg/sessionctx"
+	"github.com/pingcap/tidb/pkg/sessionctx/variable"
+	"github.com/pingcap/tidb/pkg/util/execdetails"
+	"github.com/tikv/client-go/v2/util"
+)
+
+// SlowLogFieldAccessor defines how to get or set a specific field in SlowQueryLogItems.
+// - Setter is optional and pre-fills the field before matching if it needs explicit preparation.
+// - Match evaluates whether the field in SlowQueryLogItems meets a specific threshold.
+//   - threshold is the value to compare against when determining a match.
+type SlowLogFieldAccessor struct {
+	Setter func(ctx context.Context, seCtx sessionctx.Context, items *variable.SlowQueryLogItems)
```

Comment:
`SlowLogFieldAccessor` is defined here to avoid circular imports, because `seCtx` is defined in `pkg/sessionctx/variable/slow_log.go`.